core/commands/dns.go

```diff
@@ -63,7 +63,7 @@ The resolver will give:
 			depth = namesys.DefaultDepthLimit
 		}
 		output, err := resolver.ResolveN(req.Context().Context, name, depth)
-		if err != nil {
+		if err != nil && (err != namesys.ErrResolveRecursion || depth > 1) {
```

Comment:
On Tue, Jun 09, 2015 at 04:29:15PM -0700, Juan Batiz-Benet wrote:
> > @@ -63,7 +63,7 @@ The resolver will give:
> >  			depth = namesys.DefaultDepthLimit
> >  		}
> >  		output, err := resolver.ResolveN(req.Context().Context, name, depth)
> > -		if err != nil {
> > +		if err != nil && (err != namesys.ErrResolveRecursion || depth > 1) {
> 
> ah ok. so it returns a value AND an error. can we maybe make sure
> the value is there then? bad things will happen if there is an error
> returned and a nil value.

Only if the returned error is ErrResolveRecursion.  I guess we could
add a check for that here [1] so we know we were ok by the time we got
to [2].  But I'd rather avoid even more complicated error logic in the
client code.

Alternatively, we could push the depth=1 special casing down into the
resolve() helper, but I think the current symmetry makes more sense
there and I'd rather keep the “non-recursive resolution doesn't get an
/ipfs/ path” logic up in the command implementation.

Either way, we're going to have some funkyness somewhere, so you can
also just push me in the direction you think is most reasonable.

[1]: https://github.com/ipfs/go-ipfs/blob/feba3e1da5d43684c949828c35e72256272fc7d8/namesys/base.go#L20
[2]: https://github.com/ipfs/go-ipfs/blob/feba3e1da5d43684c949828c35e72256272fc7d8/namesys/base.go#L32